src/main/java/com/artipie/http/PromuSlice.java

```diff
@@ -54,8 +51,12 @@ public Response response(final String line, final Iterable<Map.Entry<String, Str
         final Publisher<ByteBuffer> body) {
         final String ctype = new Accept(headers).values().get(0);
         try (StringWriter writer = new StringWriter()) {
-            final Set<String> filters = PromuSlice.params(line, "name");
-            final MetricsOutput output = new PrometheusOutput(writer, ctype, filters);
+            final List<? extends String> filters = new RqParams(
+                new RequestLineFrom(line).uri()
+            ).values("name");
+            final MetricsOutput output = new PrometheusOutput(
+                writer, ctype, new HashSet<>(filters)
+            );
```

Comment:
It seems we use `filters` only once, I'd suggest inline this variable

Reply:
fixed